modules/rostests/apitests/advapi32/RtlEncryptMemory.c

```diff
@@ -9,26 +9,117 @@
     ok_ntstatus(Status, STATUS_INVALID_PARAMETER);
 
-    /* Buffer must not be aligned to 8 bytes */
-    Status = RtlEncryptMemory(&Buffer[1], 8, RTL_ENCRYPT_OPTION_SAME_PROCESS);
+    /* Buffer can be unaligned */
+    RtlCopyMemory(Buffer, TestData, sizeof(Buffer));
+    Status = RtlEncryptMemory(&Buffer[1], 8, OptionFlags);
+    ok_ntstatus(Status, STATUS_SUCCESS);
+    Status = RtlDecryptMemory(&Buffer[1], 8, OptionFlags);
     ok_ntstatus(Status, STATUS_SUCCESS);
+    ok(RtlEqualMemory(Buffer, TestData, sizeof(Buffer)) == 1,
+       "OptionFlags=%lu, Buffer='%s', TestData='%s'\n",
+       OptionFlags, Buffer, TestData);
 
     RtlCopyMemory(Buffer, TestData, sizeof(Buffer));
-    Status = RtlEncryptMemory(Buffer, sizeof(Buffer), RTL_ENCRYPT_OPTION_SAME_PROCESS);
+    Status = RtlEncryptMemory(Buffer, sizeof(Buffer), OptionFlags);
     ok_ntstatus(Status, STATUS_SUCCESS);
     ok_int(RtlEqualMemory(Buffer, TestData, sizeof(Buffer)), 0);
-    Status = RtlDecryptMemory(Buffer, sizeof(Buffer), RTL_ENCRYPT_OPTION_SAME_PROCESS);
+    Status = RtlDecryptMemory(Buffer, sizeof(Buffer), OptionFlags);
+    ok_ntstatus(Status, STATUS_SUCCESS);
+    ok_int(RtlEqualMemory(Buffer, TestData, sizeof(Buffer)), 1);
+}
+
+void TestCrossProcessDecrypt(PSTR Param)
+{
+    UCHAR Buffer[32] = { 0 };
+    ULONG OptionFlags;
+    PSTR StrData;
+    ULONG i;
+    NTSTATUS Status;
+
+    printf("TestCrossProcessDecrypt(%s)\n", Param);
+
+    OptionFlags = Param[3] == '1' ? RTL_ENCRYPT_OPTION_CROSS_PROCESS : RTL_ENCRYPT_OPTION_SAME_LOGON;
+
+    /* Convert the HEX string to binary ('-cp1:<data>' */
+    StrData = Param + 5;
+    for (i = 0; i < 32; i++)
+    {
+        //Buffer[i] = ConvertHex8(StrData[2 * i], StrData[2 * i + 1]);
+        unsigned x;
+        sscanf(&StrData[2 * i], "%02X", &x);
+        Buffer[i] = (UCHAR)x;
+    }
+
+    /* Decrypt the data */
+    Status = RtlDecryptMemory(Buffer, sizeof(Buffer), OptionFlags);
     ok_ntstatus(Status, STATUS_SUCCESS);
     ok_int(RtlEqualMemory(Buffer, TestData, sizeof(Buffer)), 1);
+}
+
+void TestCrossProcessEncrypt(ULONG OptionFlags)
+{
+    UCHAR Buffer[32];
+    CHAR CmdLine[MAX_PATH];
+    PSTR StrBuffer;
+    NTSTATUS Status;
+    ULONG i;
+    INT result;
+
+    /* Encrypt the test string */
+    RtlCopyMemory(Buffer, TestData, sizeof(Buffer));
+    Status = RtlEncryptMemory(Buffer, sizeof(Buffer), OptionFlags);
+    ok_ntstatus(Status, STATUS_SUCCESS);
+
+    /* Start building a command line */
+    sprintf(CmdLine, "advapi32_apitest.exe RtlEncryptMemory -cp%lu:", OptionFlags);
+    StrBuffer = CmdLine + strlen(CmdLine);
+
+    /* Convert encrypted data into a hex string */
+    for (i = 0; i < 32; i++)
+    {
+        sprintf(&StrBuffer[2 * i], "%02X", Buffer[i]);
+    }
+
+    printf("TestCrossProcessDecrypt(%s)\n", StrBuffer - 5);
+
+    result = system(CmdLine);
+    ok_int(result, 0);
+}
+
+START_TEST(RtlEncryptMemory)
+{
+    CHAR Buffer[8] = { 0 };
```

Comment:
```suggestion
    CHAR Buffer[RTL_ENCRYPT_MEMORY_SIZE] = { 0 };
```
And use a `#define BUFFER_SIZE (4 * RTL_ENCRYPT_MEMORY_SIZE)` where it applies.